### PR TITLE
Dialog parameter

### LINKED ISSUE #1471: core: frontend: parameter editor doing weird things with bitmasks
### Current behaviour

There seems to be some kind of extra phantom bits being displayed (they don't appear in QGC):

https://user-images.githubusercontent.com/25898329/218676810-57981881-e028-48ee-9332-8f749ba786a3.mov

### Expected or desired behaviour

It should only show the bits that are actually set, and should not accidentally set any bits that were not requested.

### Prerequisites

- [X] I have checked to make sure that a similar request has not already been filed or fixed.

### PR DESCRIPTION
**Pull request by patrickelectric**
Move parameter dialog to its own component to be used in different places.
Also turns the `autopilot reboot` button to be global, to follow the user between the pages until it's rebooted. 
Peek 13-02-2023 18-45

Fix #1471

### CODE REVIEW

File: core/frontend/src/components/parameter-editor/ParameterEditor.vue [hunk 1]
```diff
@@ -329,96 +211,18 @@ export default Vue.extend({
...
+      console.log('edit!', this.edit_dialog)
       if (param.readonly) {
+        console.log('read only!')
         return
       }
       this.edited_param = param
-      this.new_value = this.edited_param?.value ?? 0
+      // this.new_value = this.edited_param?.value ?? 0
       this.edit_dialog = true
+      console.log('>>>', this.edit_dialog)
```
Comment (Williangalvani) [inline core/frontend/src/components/parameter-editor/ParameterEditor.vue:225]: debug logs and commented code

File: core/frontend/src/components/parameter-editor/ParameterEditorDialog.vue [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+                  :rules="[() => true]"
+                />
+                <v-text-field
+                  v-if="
+                    custom_input ||
+                      (param
+                        && !param.options
+                        && !param.bitmask
+                      )
+                  "
```
Comment (Williangalvani) [inline core/frontend/src/components/parameter-editor/ParameterEditorDialog.vue:56]: I didn't know you could do multiline strings here, cool

**Review by Williangalvani — CHANGES_REQUESTED** [review]

**Review by ES-Alexander — APPROVED** [review]
This looks great - love it! :-)

Is the blinking reboot icon tied to pirate mode somehow, or are the pirate mode mentions in the tray menu stuff just to get the icon to show up in the header? The parameter editor is available in friendly mode, so the reboot icon should be too (it might already be fine) :-)

File: core/frontend/src/components/app/VehicleRebootMenu.vue [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import settings from '@/libs/settings'
+import autopilot_data from '@/store/autopilot'
+import autopilot from '@/store/autopilot_manager'
+import { parameters_service } from '@/types/frontend_services'
+import back_axios from '@/utils/api'
+
+const notifier = new Notifier(parameters_service)
+
+export default Vue.extend({
+  name: 'PirateModeMenu',
```
Comment (Williangalvani) [inline core/frontend/src/components/app/VehicleRebootMenu.vue:36]:
```suggestion
  name: 'VehicleRebootMenu',
```

File: core/frontend/src/components/app/VehicleRebootMenu.vue [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  methods: {
+    async rebootVehicle(): Promise<void> {
+      this.$emit('vehicleRebootCalled', settings.is_pirate_mode)
```
Comment (Williangalvani) [inline core/frontend/src/components/app/VehicleRebootMenu.vue:39]: this does look off

Comment (patrickelectric) [reply core/frontend/src/components/app/VehicleRebootMenu.vue:39]: It was for testing, removing

**Comment by patrickelectric** [comment]
> This looks great - love it! :-)
> 
> Is the blinking reboot icon tied to pirate mode somehow, or are the pirate mode mentions in the tray menu stuff just to get the icon to show up in the header? The parameter editor is available in friendly mode, so the reboot icon should be too (it might already be fine) :-)

It was something that I did for internal test, forgot to remove it

**Comment by Williangalvani** [comment]
can you fix this issue? when we type a number it immediately switches to the option:

https://user-images.githubusercontent.com/4013804/219120163-bacbbb35-15bb-489a-98d0-971e0873a06d.mov

it is pre-existent, but since you are working in this component =P

**Comment by patrickelectric** [comment]
@Williangalvani check now

File: core/frontend/src/components/parameter-editor/ParameterEditorDialog.vue [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+    },
+    edited_bitmask(): number[] {
+      const edited_bitmask = [] as number[]
+
+      if (this.param.readonly || this.param.bitmask === undefined) {
+        return edited_bitmask
+      }
+
+      // eslint-disable-next-line no-bitwise
+      for (let v = 0; 1 << v < this.param.value; v += 1) {
```
Comment (Williangalvani) [inline core/frontend/src/components/parameter-editor/ParameterEditorDialog.vue]: does this work for v>32?

**Review by Williangalvani — APPROVED** [review]
nice!

**Comment by ES-Alexander** [comment]
> it is pre-existent, but since you are working in this component =P

Lol, nice logic - shame I didn't think to recommend #1210 at the same time...

**Comment by patrickelectric** [comment]
> > it is pre-existent, but since you are working in this component =P
> 
> Lol, nice logic @Williangalvani - shame I didn't think to recommend #1210 at the same time...

I'm not a fan of changing the scope of a PR since this can result on bigger changes and drag the PR merge time with it. That's one of the reasons that ArduPilot has a PR list hell.

**Pull request merged**